packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -287,4 +298,21 @@ describe('<XGrid /> - Events Params', () => {
     gridWindow.dispatchEvent(new Event('scroll'));
     expect(handleOnRowsScrollEnd.callCount).to.equal(1);
   });
+
+  it('call onViewportRowChange when the viewport rows change', () => {
+    const handleOnVirtualPageChange = spy();
+    const { container } = render(
+      <div style={{ width: 300, height: 300 }}>
+        <TestVirtualization onViewportRowChange={handleOnVirtualPageChange} />
+      </div>,
+    );
+    const gridWindow = container.querySelector('.MuiDataGrid-window');
+    // scroll 10 rows.
+    gridWindow.scrollTop = 12345;
+    gridWindow.dispatchEvent(new Event('scroll'));
+
+    expect(handleOnVirtualPageChange.callCount).to.equal(1);
+    expect(handleOnVirtualPageChange.lastCall.args[0].firstRowIndex).to.equal(48);
+    expect(handleOnVirtualPageChange.lastCall.args[0].lastRowIndex).to.equal(50);
```

Comment:
This looks suspicious. The default row height is 52 pixels. We have 300 pixels in height, so space for 5.7 rows. Removing the header, I would expect that we get 
```suggestion
    expect(handleOnVirtualPageChange.lastCall.args[0].firstRowIndex).to.equal(46 = 50 - 1 - Math.ceil((300 - 52 - 2)/52));
    expect(handleOnVirtualPageChange.lastCall.args[0].lastRowIndex).to.equal(50);
```

Reply:
But that event is still being emitted when the virtualization changes, not when you scroll. Basically, it won't emit values such as `0-4`, `1-5`, `2-6` but rather `6-10`, `10-14`, `14-18` etc.

I think we should first agree on the behaviour we would want.

Reply:
> I think we should first agree on the behaviour we would want.

We have started this discussion in https://github.com/mui-org/material-ui-x/pull/1715#issuecomment-866723993. I didn't realize this wasn't clear yet. I mean, I thought that I answered the question and that you upvoted to confirm.

Reply:
Ok, not a problem, I was just still thinking in the context of the virtualization.

Reply:
Updated